src/core/qgsrelationmanager.cpp

```diff
@@ -93,6 +93,19 @@ void QgsRelationManager::removeRelation( const QgsRelation &relation )
 
 QgsRelation QgsRelationManager::relation( const QString &id ) const
 {
+  if ( !mRelations.contains( id ) )
+  {
+    // Check whether the provided ID refers to a polymorphic relation's generated ID using QGIS < 3.38
```

Comment:
```suggestion
    // Check whether the provided ID refers to a polymorphic relation's generated ID using QGIS < 3.38
    // If it has not yet been upgraded to a stable id, upgrade now
```

I think that gives a bit more context